src/tagifyme/model/Database.java

```diff
@@ -197,24 +198,56 @@ public Iterable<Pair<Data, Set<Tag>>> data_and_tags() {
     return r;
   }
   
-  public Iterable<Tag> tags(){
+  /**
+   * Return the `Tag`s within the `Database`.
+   */
+  public Iterable<Tag> allTags(){
       return tag_set;
   }
-
   
+  // TODO: I don't know why this is needed.
+  // This might just be a `contains`?
   public Tag getTag(String name){
     int i = 0;
     for (Tag tag : tag_set ){
         if(tag.getName().equals(name)){
             return tag;
         };
     }
-    return noTag;
+    return new Tag("Undefined");
   }
   
+  /**
+   * Delete some Data from the Database by PATH.
+   */
+  public void deleteDataByPATH(String PATH) {
+    // TODO: I have no idea why you've chosen to delete
+    // via PATH; it's likely easier to do it with `name`
```

Comment:
When I decided to use PATH to delete I figured it was because we could have duplicate names. I realize now that is not the case